packages/amplify-e2e-tests/src/init-special-cases/index.ts

```diff
@@ -0,0 +1,85 @@
+import fs from 'fs-extra';
+import path from 'path';
+import os from 'os';
+import { nspawn as spawn, getCLIPath } from 'amplify-e2e-core';
+
+export async function initWithoutCredentialFileAndNoNewUserSetup(projRoot) {
+  const settings = {
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID,
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    region: process.env.AWS_REGION || process.env.AWS_DEFAULT_REGION || 'us-west-2',
+  };
+
+  delete process.env.AWS_ACCESS_KEY_ID;
+  delete process.env.AWS_SECRET_ACCESS_KEY;
+  delete process.env.AWS_DEFAULT_REGION;
+  delete process.env.AWS_REGION;
+
+  const dotAWSDirPath = path.normalize(path.join(os.homedir(), '.aws'));
+  const credentialsFilePath = path.join(dotAWSDirPath, 'credentials');
+  const configFilePath = path.join(dotAWSDirPath, 'config');
+  const credentialsFilePathHide = path.join(dotAWSDirPath, 'credentials.hide');
+  const configFilePathHide = path.join(dotAWSDirPath, 'config.hide');
+
+  try {
+    if (fs.existsSync(configFilePath)) {
+      fs.renameSync(configFilePath, configFilePathHide);
+    }
+    if (fs.existsSync(credentialsFilePath)) {
+      fs.renameSync(credentialsFilePath, credentialsFilePathHide);
+    }
+    await initWorkflow(projRoot, settings);
+  } catch (e) {
+    throw e;
+  } finally {
+    if (fs.existsSync(configFilePathHide)) {
+      fs.renameSync(configFilePathHide, configFilePath);
+    }
+    if (fs.existsSync(credentialsFilePathHide)) {
+      fs.renameSync(credentialsFilePathHide, credentialsFilePath);
+    }
+  }
+}
```

Comment:
the better approach here is to use mock/spy instead of renaming the files, which makes one less thing to ensure.

See if you can use 
```js
const existSyncOrig = fs.existsSync;
jest.spyOn(fs, 'existsSync').mockImplementation((path) => {
   return [credentialsFilePath, configFilePath].includes(path) ? false :
 existSyncOrig(path)});

```

Reply:
I tried different combinations, but using jest.spyOn does not work, seems like jest mock can not penetrate into the spawned process. 
I updated the test to use another mechanism to avoid having to rename the credential files.

Reply:
I would rather do the first approach then, instead of using a custom script to invoke the CLI

Reply:
Agree, got back to the original test setup.